### ISSUE TITLE
layer surface 'on_demand' keyboard focus does not receive focus until the user clicks on it.

### ISSUE THREAD

**Issue by marcusbritanicus** [WayfireWM/wayfire#1276]
**Describe the bug**
If we have a client that uses layer shell protocol, and set the keyboard focus to `on_demand`, it does not get the keyboard focus until the user clicks on it.

**To Reproduce**
1. Open a client that uses layer shell protocol and requires `on_demand` keyboard focus.
2. You'll see that it does not have keyboard focus, until you click it with a mouse pointer.

**Expected behavior**
Give keyboard focus to the layer surface as soon as it is mapped.

**Wayfire version**
git

**PS:** This was tested using a Qt app. It is possible that this bug is due to Qt's bad implementation of things.

**Comment by marcusbritanicus** [comment]
I can confirm that this is a wayfire bug and not due to Qt. I tested my drop-down terminal with `on_demand` keyboard focus on sway. It receives keyboard focus as soon as it is shown.

**Comment by ammen99** [comment]
We probably need the same as #1933 in the map handler of layer-shell views.
